lib/cli/kit/command_registry.rb

```diff
@@ -0,0 +1,79 @@
+require 'cli/kit'
+
+module CLI
+  module Kit
+    module CommandRegistry
+      attr_accessor :commands, :aliases
+      class << self
+        attr_accessor :registry_target
+      end
+
+      def resolve_contextual_command
```

Comment:
The concept of "Project-local" and "Type" commands was generalized a bit into "Contextual" commands. The interface to add contextual commands to a command registry is:

* `resolve_contextual_command`
* `contextual_aliases`
* `contextual_command_class`

This is a pretty niche concern and I don't really expect to build too many tools other than `dev` that use this feature.

Reply:
So this is only included for stuff from a `dev.yml`?

Reply:
Well it's "included" everywhere depending on what you mean, but, well: https://github.com/Shopify/dev/blob/4eea78118cc0f6fe4180f37aef97ab676101915c/lib/dev/commands.rb#L7-L19